packages/gatsby/src/query/graphql-errors.js

```diff
@@ -221,3 +217,82 @@ export function graphqlError(
 
   return { formattedMessage: reportedMessage, docName, message, codeBlock }
 }
+
+export function unknownFragmentError({ fragmentNames, filePath, def, node }) {
+  const name = node.name.value
+  const closestFragment = fragmentNames
+    .map(f => {
+      return { fragment: f, score: levenshtein.get(name, f) }
+    })
+    .filter(f => f.score < 10)
+    .sort((a, b) => a.score > b.score)[0]?.fragment
+
+  return {
+    id: `85908`,
+    filePath,
+    context: {
+      fragmentName: name,
+      closestFragment,
+      codeFrame: codeFrameColumns(
+        def.text,
+        {
+          start: getLocation({ body: def.text }, node.loc.start),
+          end: getLocation({ body: def.text }, node.loc.end),
+        },
+        {
+          linesAbove: 10,
+          linesBelow: 10,
+        }
+      ),
+    },
+  }
+}
+
+export function duplicateFragmentError({ leftDefinition, rightDefinition }) {
```

Comment:
if should be added name here:
```suggestion
export function duplicateFragmentError({ name, leftDefinition, rightDefinition }) {
```